.github/workflows/ci-main.yml

```diff
@@ -45,23 +45,39 @@ jobs:
         with:
           submodules: "recursive"
 
+# wesnothd and campaignd should be buildable without SDL2 being present
       - name: Build wesnoth, wesnothd, campaignd and unit tests
         id: build # needed to check step outcome
         run: |
           case $TOOL in
             scons)
-              scons wesnoth wesnothd campaignd boost_unit_tests \
+              scons wesnoth boost_unit_tests \
+                build="$CFG" ctool="$CC" cxxtool="$CXX" cxx_std="$CXX_STD" \
+                extra_flags_config="-pipe" strict=true forum_user_handler=true \
+                nls=false enable_lto="$LTO" force_color=true jobs=2 --debug=time
+              
+              apt remove -y -qq libsdl2-dev libsdl2-dev libsdl2-mixer-dev
+
+              scons wesnothd campaignd \
                 build="$CFG" ctool="$CC" cxxtool="$CXX" cxx_std="$CXX_STD" \
                 extra_flags_config="-pipe" strict=true forum_user_handler=true \
                 nls=false enable_lto="$LTO" force_color=true jobs=2 --debug=time
               ;;
             cmake)
-              cmake -DENABLE_GAME=true -DENABLE_SERVER=true -DENABLE_CAMPAIGN_SERVER=true \
+              cmake -DENABLE_GAME=true -DENABLE_SERVER=false -DENABLE_CAMPAIGN_SERVER=false \
                 -DENABLE_TESTS=true -DCMAKE_BUILD_TYPE="$CFG" -DCXX_STD="$CXX_STD" \
                 -DEXTRA_FLAGS_CONFIG="-pipe" -DENABLE_STRICT_COMPILATION=true -DENABLE_MYSQL=true \
                 -DENABLE_NLS=false -DFORCE_COLOR_OUTPUT=true -DENABLE_LTO="$LTO" -DLTO_JOBS=2 .
               make conftests
               make VERBOSE=1 -j2
+
+              apt remove -y -qq libsdl2-dev libsdl2-dev libsdl2-mixer-dev
```

Comment:
The 2nd `libsdl2-dev` should probably have been libsdl2-image-dev

Reply:
Just noting that this same error appears both in the `scons` and the `cmake` builds.

Reply:
Actually, since libsdl2-image-dev and libsdl2-mixer-dev both depend on libsdl2-dev, it doesn't make a difference if they are listed here.

Reply:
Fixed.